nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -489,3 +490,268 @@ def _parse_item(self, line: str, manifest_file: str) -> Dict[str, Any]:
+
+        Args:
+            audio_files:
+                List of audio filepaths.
+            durations:
+                List of float durations.
+            rttm_files:
+                List of RTTM files (Groundtruth diarization annotation file).
+            offsets:
+                List of offsets or None.
+            target_spks (tuple):
+                List of tuples containing the two inndices of targeted speakers for evaluation.
+                Example: [(0, 1), (1, 2), (2, 3)]
+            sess_spk_dict (Dict):
+                List of Mapping dictionaries between RTTM speakers and speaker labels in the clustering result.
+            clus_spk_digits (tuple):
+                List of Tuple containing all the speaker indices from the clustering result.
+                Example: [(0, 1, 2, 3), (0, 1, 2), ...]
+            rttm_spkr_digits (tuple):
+                List of tuple containing all the speaker indices in the RTTM file.
+                Example: (0, 1, 2), (0, 1), ...]
+            max_number: Maximum number of samples to collect
+            do_sort_by_duration: True if sort samples list by duration
+            index_by_file_id: If True, saves a mapping from filename base (ID) to index in data.
+        """
+
+        if index_by_file_id:
+            self.mapping = {}
+        output_type = self.OUTPUT_TYPE
+        data, duration_filtered = [], 0.0
+
+        zipped_items = zip(
+            audio_files, durations, rttm_files, offsets, target_spks_list, sess_spk_dicts, clus_spk_list, rttm_spk_list
+        )
+        for (
+            audio_file,
+            duration,
+            rttm_file,
+            offset,
+            target_spks,
+            sess_spk_dict,
+            clus_spk_digits,
+            rttm_spk_digits,
+        ) in zipped_items:
+
+            if duration is None:
+                duration = 0
+
+            data.append(
+                output_type(
+                    audio_file,
+                    duration,
+                    rttm_file,
+                    offset,
+                    target_spks,
+                    sess_spk_dict,
+                    clus_spk_digits,
+                    rttm_spk_digits,
+                )
+            )
+
+            if index_by_file_id:
+                file_id, _ = os.path.splitext(os.path.basename(audio_file))
+                self.mapping[file_id] = len(data) - 1
+
+            # Max number of entities filter.
+            if len(data) == max_number:
+                break
+
+        if do_sort_by_duration:
+            if index_by_file_id:
+                logging.warning("Tried to sort dataset by duration, but cannot since index_by_file_id is set.")
+            else:
+                data.sort(key=lambda entity: entity.duration)
+
+        logging.info(
+            "Filtered duration for loading collection is %f.", duration_filtered,
+        )
+        logging.info(f"Total {len(data)} session files loaded accounting to")
```

Comment:
accounting to ....?

Reply:
Adding "len(audio_files)" of audio clips.